python-spec/src/somacore/query/query.py

```diff
@@ -434,6 +501,59 @@ def _axisp_inner(
         joinids = getattr(self._joinids, axis.value)
         return axisp[layer].read((joinids, joinids))
 
+    def _axism_inner(
+        self,
+        axis: "_Axis",
+        layer: str,
+    ) -> data.SparseRead:
+        key = axis.value + "m"
+
+        if key not in self._ms:
+            raise ValueError(f"Measurement does not contain {key} data")
+
+        axism = self._ms.obsm if axis is _Axis.OBS else self._ms.varm
+        if not (layer and layer in axism):
+            raise ValueError(f"Must specify '{key}' layer")
+        
+        if not isinstance(axism[layer], data.SparseNDArray):
+            raise TypeError(f"Unexpected SOMA type stored in '{key}' layer")
+
+        joinids = getattr(self._joinids, axis.value)
+        return axism[layer].read((joinids, slice(None)))
+    
+    def _convert_to_ndarray(self, is_obs: bool, T: pa.Table, n_row: int, n_col: int) -> np.ndarray:
```

Comment:
I'm fairly sure that this has a bug -- it always reindexes the obs joinids, even if axis is VAR.  And it calculates coordinates with the var dimension even when it shouldn't 

you probably want something like:
```
major_dim = 0 if is_obs else 1
minor_dim = 1-major_dim
idx = (self.indexer.by_obs if is_obs else self.indexer.by_var)(T[f"soma_dim_{major_dim}"])
Z = np.zeros(n_row * n_col, dtype=np.float32)
np.put(Z, idx * n_col + T[f"soma_dim_{minor_dim}"], T["soma_data"])
return Z.reshape(n_row, n_col)
```

Reply:
oh, and good to ensure that the unit tests are sufficient to capture this bug ;-)

Reply:
update:  reviewer (me) confusion. Apologies.